crates/chain/src/keychain/txout_index.rs

```diff
@@ -588,3 +588,63 @@ where
         .take_while(Result::is_ok)
         .map(Result::unwrap)
 }
+
+pub struct SpkIterator<'a> {
+    next_index: u32,
+    end: u32,
+    descriptor: Cow<'a, Descriptor<DescriptorPublicKey>>,
+    secp: Secp256k1<bitcoin::secp256k1::VerifyOnly>,
+}
+
+impl<'a> SpkIterator<'a> {
+    /// Creates a new script pubkey iterator starting at 0 from a descriptor
+    pub fn new(descriptor: Cow<'a, Descriptor<DescriptorPublicKey>>) -> Self {
```

Comment:
Why does the `descriptor` param need to be a `Cow<>` ? I've never used a `Cow<>` but from the docs it looks like it's needed if `Descriptor<DescriptorPublicKey>>` is mutable and I don't see here where it needs to be.

Reply:
You can disregard my comment above. I think we need the `Cow<>` so that  `SpkIterator` will be Send+Sync, but I still need to verify it solves my issue in #894.

Reply:
I used `Cow` because the iterator is used for both internal logic (where we don't want to do an unnecessary clone of the descriptor), and also returned (where we do want to clone the descriptor as we don't want to hold a reference to our structures).

Reply:
> You can disregard my comment above. I think we need the `Cow<>` so that `SpkIterator` will be Send+Sync, but I still need to verify it solves my issue in #894.

It only needs to be `Send + 'static` I think. What do we need `Sync` for? It's only being accessed from one thread. We need to remove the `Cow` so it can be `'static`. Can't this just take `T: Borrow<Descriptor<DescriptorPublicKey>>` or something like that.

Reply:
With @evanlinjin's assistance, `Cow<>` has removed in lieu of `T: Borrow<Descriptor<DescriptorPublicKey>>`.

Reply:
> > You can disregard my comment above. I think we need the `Cow<>` so that `SpkIterator` will be Send+Sync, but I still need to verify it solves my issue in #894.
> 
> It only needs to be `Send + 'static` I think. What do we need `Sync` for? 
Agreed, doesn't need to be `Sync`.  And make sense to use `Borrow<>` here.